charts/metrics-server/templates/deployment.yaml

```diff
@@ -11,6 +11,7 @@ metadata:
   {{- end }}
 spec:
   replicas: {{ .Values.replicas }}
+  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
```

Comment:
```suggestion
  {{- if or .Values.revisionHistoryLimit (eq .Values.revisionHistoryLimit 0) }}
  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
  {{- end }}
```

Reply:
This works fine if it's unset, or set to 10 (for example). If I set it to 0, then I get:
```
Error: template: metrics-server/templates/deployment.yaml:14:42: executing "metrics-server/templates/deployment.yaml" at <eq .Values.revisionHistoryLimit 0>: error calling eq: incompatible types for comparison

Use --debug flag to render out invalid YAML
```

Reply:
Did you test it with `0` or `"0"`, the latter would be expected to error? I've tested it successfully.

Reply:
`0`, no quotes. I'm testing by running `helm template`, changing the values.yaml and checking the rendered output. I'm using helm 3.13.2.

Reply:
I'm guessing you were using `--set revisionHistoryLimit=0` to test?

After doing a bit of experimenting, I think we're actually running into https://github.com/kubernetes-sigs/yaml/issues/45 here. If I template in `{{ typeOf .Values.revisionHistoryLimit }}` somewhere then when the value comes from `values.yaml` then it outputs `float64`, and when I use `--set` it outputs `int64` instead.

Your suggested change works fine when using `--set` and the type is `int64`, it fails when the value comes from YAML and the type is `float64`.

I've found the following handles both cases:
```mustache
{{- if or (kindIs "float64" .Values.revisionHistoryLimit) (kindIs "int64" .Values.revisionHistoryLimit) }}
revisionHistoryLimit: {{ .Values.revisionHistoryLimit | int64 }}
{{- end }}
```
If it's unset then nothing is templated in. WDYT?

Reply:
> I'm guessing you were using --set revisionHistoryLimit=0 to test?

I was indeed.

> ```yaml
> {{- if or (kindIs "float64" .Values.revisionHistoryLimit) (kindIs "int64" .Values.revisionHistoryLimit) }}
> revisionHistoryLimit: {{ .Values.revisionHistoryLimit | int64 }}
> {{- end }}
> ```

This looks like a good pattern to me.